sipi/scripts/upload.lua

```diff
@@ -199,19 +218,77 @@ for file_index, file_params in pairs(server.uploads) do
     --
     -- prepare and write sidecar file
     --
-    local sidecar_data = {
-        originalFilename = original_filename,
-        checksumOriginal = checksum_original,
-        originalInternalFilename = hashed_tmp_storage_original,
-        internalFilename = tmp_storage_filename,
-        checksumDerivative = checksum_derivative
-    }
+    local sidecar_data = {}
+
+    if media_type == VIDEO then
+        
+        local handle
+        -- get video file information with ffprobe: widht, height, duration and frame rate (fps)
+        handle = io.popen("ffprobe -v error -select_streams v:0 -show_entries stream=width,height,bit_rate,duration,nb_frames,r_frame_rate -print_format json -i " .. tmp_storage_file_path)
+        local file_meta = handle:read("*a")
+        handle:close()
+        -- decode ffprobe output into json, but only first stream
+        local file_meta_json = json.decode( file_meta )['streams'][1]
+        
+        -- get video duration
+        local duration = tonumber(file_meta_json['duration'])
+        if not duration then
+            send_error(417, "upload.lua: ffprobe get duration failed: " .. duration)
+        end
+        -- get video width (dimX)
+        local width = tonumber(file_meta_json['width']);
+        if not width then
+            send_error(417, "upload.lua: ffprobe get width failed: " .. width)
+        end
+        -- get video height (dimY)
+        local height = tonumber(file_meta_json['height'])
+        if not height then
+            send_error(417, "upload.lua: ffprobe get height failed: " .. height)
+        end
+        -- get video fps
+        -- this is a bit tricky, because ffprobe returns something like 30/1 or 179/6; so, we have to convert into a floating point number;
+        -- or we can calculate fps from number of frames divided by duration
+        local fps
+        local frames = tonumber(file_meta_json['nb_frames'])
+        if not frames then
+            send_error(417, "upload.lua: ffprobe get frames failed: " .. frames)
+        else
+            fps = frames / duration
+            if not fps then
+                send_error(417, "upload.lua: ffprobe get fps failed: " .. fps)
+            end
+        end
+
+        sidecar_data = {
+            originalFilename = original_filename,
+            checksumOriginal = checksum_original,
+            originalInternalFilename = hashed_tmp_storage_original,
+            internalFilename = tmp_storage_filename,
+            checksumDerivative = checksum_derivative,
+            width = width,
+            height = height,
+            duration = duration,
+            fps = fps
+        }
+
+        -- TODO: similar setup for audio files; get duration with ffprobe and write extended sidecar data
```

Comment:
Should this be kept here? Is there a JIRA issue defined for this task or how do we remember to actually do it?

Reply:
This is the question. I've created a new subtask DEV-770